sstables/sstable_directory.cc

```diff
@@ -453,6 +457,11 @@ future<> sstable_directory::restore_components_lister::process(sstable_directory
                 return directory._storage_opts->append_to_s3_prefix(prefix);
             });
     });
+    co_await smp::submit_to(0, [this] {
+        // increment the number of processed sstables, this number is used by
+        // the download_task_impl for tracking the task progress
+        *_num_processed += 1;
```

Comment:
Patch comment says

> The restore process happens in two sequential steps:
> 1. Download and open specified SSTables from object storage
> 2. Stream mutation fragments from downloaded SSTables to mapped destinations

I find the word "download" in item 1 misleading. At this stage only summary files are opened and read, no data transfer happens. And while technically it's indeed "downloading" if a user has, say, 1TB sstable in a bucket, phase 1 will complete _much_ faster than it takes do actually download 1TB.

Having said that, this patch only tracks a tiny portion of the overall process and doesn't seem to be enough (so the "fixes" tag is not correct either)

Reply:
@xemul so, IIUC, what you imply is to to track the progress load-and-stream step, am i right?

Reply:
Yes, I think this is the largest time-taking part of the restore, and it should be progress-tracked

Reply:
thank you for confirming this. will implement it in the next revision.